string message_id
int64 message_data1
float64[] message_data2